v2/app/src/main/java/com/google/firebase/example/makeitso/ui/settings/SettingsViewModel.kt

```diff
@@ -0,0 +1,39 @@
+package com.google.firebase.example.makeitso.ui.settings
+
+import com.google.firebase.example.makeitso.MainViewModel
+import com.google.firebase.example.makeitso.data.repository.AuthRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import javax.inject.Inject
+
+@HiltViewModel
+class SettingsViewModel @Inject constructor(
+    private val authRepository: AuthRepository
+) : MainViewModel() {
+    private val _isAnonymous = MutableStateFlow(true)
+    val isAnonymous: StateFlow<Boolean>
+        get() = _isAnonymous.asStateFlow()
+
+    fun loadCurrentUser() {
+        launchCatching {
+            val currentUser = authRepository.currentUser
+            _isAnonymous.value = currentUser != null && currentUser.isAnonymous
+        }
+    }
+
+    fun signOut(openHomeScreen: () -> Unit) {
```

Comment:
I don't like the idea of passing callbacks to the ViewModel. I think `openHomeScreen()` should be called by the UI layer, not the ViewModel. The ViewModel should simply handle signing out the user. The UI is responsible for whatever happens after the user is signed out.

Reply:
Updated navigation logic according to the Android architecture!